tests/test_process.py

```diff
@@ -604,29 +599,62 @@ def test_process_index(tmpdir, num_workers, multiprocessing):
 
     # segmented index with relative paths
     index = audformat.segmented_index(
-        [file] * 3,
-        pd.timedelta_range('0s', '2s', 3),
-        pd.timedelta_range('1s', '3s', 3),
+        [file] * 4,
+        starts=[0, 0, 1, 2],
+        ends=[None, 1, 2, 3],
+    )
+    y = process.process_index(
+        index,
+        preserve_index=preserve_index,
+        root=root,
     )
-    y = process.process_index(index, root=root)
     for (file, start, end), value in y.items():
         signal, sampling_rate = audinterface.utils.read_audio(
             file, start=start, end=end, root=root
         )
         np.testing.assert_equal(signal, value)
 
-    # filewise index with relative paths
+    # filewise index with absolute paths
     index = audformat.filewise_index(path)
-    y = process.process_index(index, root=root)
-    for (file, start, end), value in y.items():
-        signal, sampling_rate = audinterface.utils.read_audio(
-            file, start=start, end=end, root=root
-        )
-        np.testing.assert_equal(signal, value)
+    y = process.process_index(
+        index,
+        preserve_index=preserve_index,
+    )
+    if preserve_index:
+        for path, value in y.items():
```

Comment:
Maybe add `pd.testing.assert_frame_equal(y.index, index)` here and in the other places?

Reply:
Sorry, I mean `assert_index_equal()` of course :)

Reply:
I added it.